src/svd/parsing.py

```diff
@@ -46,7 +46,7 @@ class Options:
 
     # Make the addressOffset in cluster elements be relative to the immediate parent
     # element instead of relative to the containing peripheral.
-    parent_relative_cluster_address: bool = False
+    parent_relative_cluster_address: bool = True
```

Comment:
I'm actually wondering whether this option needs to exist at all. Have you ever encountered any SVD files in the wild which would require this to be false?

I understand that the default "false" matches the specification literally:
https://open-cmsis-pack.github.io/svd-spec/main/elem_registers.html#elem_cluster

> A \<cluster\> specifies the addressOffset relative to the baseAddress of the grouping element. All \<register\> elements within a \<cluster\> specify their addressOffset relative to the cluster base address (\<peripheral.baseAddress\> + \<cluster.addressOffset\>).

However, I have reasons to doubt that this is de facto correct. For one thing, Nordic MDK headers seem to be generated using `svdconv`, the CMSIS tool designed for this purpose. I played around with it today and found that it takes the addressOffset of a cluster to be relative to its parent element - not necessarily the peripheral. I even found some code responsible for the calculation:
https://github.com/Open-CMSIS-Pack/devtools/blob/tools/svdconv/3.3.47/tools/svdconv/SVDModel/src/SvdItem.cpp#L698
https://github.com/Open-CMSIS-Pack/devtools/blob/tools/svdconv/3.3.47/tools/svdconv/SVDModel/include/SvdCluster.h#L32
https://github.com/Open-CMSIS-Pack/devtools/blob/tools/svdconv/3.3.47/tools/svdconv/SVDModel/src/SvdCluster.cpp#L48

It's possible that this discrepancy stems from the fact that SVD didn't always support nested clusters:

> Since version 1.3 of the specification, the nesting of \<cluster\> elements is supported. Nested clusters express hierarchical structures of registers. It is predominantely targeted at the generation of device header files to create a C-data structure within the peripheral structure instead of a flat list of registers.

Maybe after this was added, no one thought to reword the rest of the text?

Reply:
> Have you ever encountered any SVD files in the wild which would require this to be false?

No, and a [search on github for nested clusters](https://github.com/search?q=path%3A*.svd+%22%3C%2Fcluster%22&type=code) yielded only examples from the test cases of other SVD parsers :smile:

> I played around with it today and found that it takes the addressOffset of a cluster to be relative to its parent element - not necessarily the peripheral.

Good find - I'm sufficiently convinced that we can remove this option instead.

Reply:
Anyway, we might want to keep the option for backwards compatibility for now since it is used in some places, and then remove it once suitable